### PR TITLE
Licensing changes patched in.

### PR DESCRIPTION
**Pull request by Vindexus**
Bill and I did rebase stuff for this pull request.

### CODE REVIEW

File: src/js/declare.js [hunk 1]
```diff
@@ -258,6 +269,9 @@ function stateChange (url, state) {
 }
 
 function openRoute (target, container, matches) {
+  if(target.hasAttribute('redirect')) {
```
Comment (tmathews) [inline src/js/declare.js]: Please remove old code.

File: src/js/ui.js [hunk 2]
```diff
@@ -46,9 +46,13 @@ function openModal (name, data) {
 
 function closeModal () {
   var container = document.querySelector('[role="modals"]')
-  container.classList.remove('open')
-  var x = container.querySelector('[role="container"]').firstElementChild
-  x.parentElement.removeChild(x)
+  if(container != null) {
```
Comment (tmathews) [inline src/js/ui.js]: Better style is to use `if (!container) return;`

Comment (jb55) [inline src/js/ui.js]: not quite, `!= null` is a bit safer to protect against 0s, empty strings, etc

Comment (jb55) [inline src/js/ui.js]: actually what I said isn't really true, and only makes sense for `== null` checks

Comment (tmathews) [inline src/js/ui.js]: I thought that's what you were getting at. JS comparison ftw.

**Pull request merged**